File: By-Swift/By-Swift/Programmers/[2019-카카오-공채]/실패율.swift
//
//  실패율.swift
//  Algorithm-Swift
//
//  Created by 이원석 on 2021/04/16.
//
import Foundation

func solution(_ N:Int, _ stages:[Int]) -> [Int] {
    var failure = [Int:Double]()
    var people = stages.count
    
    for i in 0..<N{
      let fail = stages.filter{$0 == i+1}.count
      failure[i] = Double(fail) / Double(people)
      people -= fail
    }
    
    let answer = failure.sorted(by: <).sorted{$0.value > $1.value}.map{$0.key+1} 
    return answer
}

File: By-Python/programmers/[2019-KAKAO-BLIND-RECRUITMENT]/실패율.py
# 실패율

def solution(N, stages):
    failure = []
    people = len(stages)
    for i in range(N):
        fail = stages.count(i+1)
        if fail == 0:
            failure.append([i,0])
        else:
            failure.append([i,(fail/people)])
        people -= fail        
    
    failure = sorted(failure, key=lambda x: (-x[1],x[0]))
    return [i[0]+1 for i in failure]